src/main/java/org/opensearch/neuralsearch/processor/rerank/RerankProcessor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Aryn
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.opensearch.neuralsearch.processor.rerank;
+
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+
+public interface RerankProcessor extends SearchResponseProcessor {
```

Comment:
hmm. but I should rename `generateScoringContext` to `generateRerankingContext`, huh

Reply:
ooh @martin-gaievski separating out RescoringRerankProcessor to its own layer also lets us implement a ScorCombinationRerankProcessor that uses the score combination stuff instead of simply overwriting the scores. Not sure how that fits in the API though